metaflow/plugins/kfp/kfp.py

```diff
@@ -453,24 +467,28 @@ def _step_cli(self, node: DAGNode, task_id: str, user_code_retries: int) -> str:
 
     @staticmethod
     def _set_container_settings(container_op: ContainerOp, kfp_component: KfpComponent):
-        resource_requirements: Dict[str, str] = kfp_component.resource_requirements
-        if "memory" in resource_requirements:
-            container_op.container.set_memory_request(resource_requirements["memory"])
-        if "memory_limit" in resource_requirements:
-            container_op.container.set_memory_limit(
-                resource_requirements["memory_limit"]
-            )
-        if "cpu" in resource_requirements:
-            container_op.container.set_cpu_request(resource_requirements["cpu"])
-        if "cpu_limit" in resource_requirements:
-            container_op.container.set_cpu_limit(resource_requirements["cpu_limit"])
-        if "gpu" in resource_requirements:
+        pod_spec: Dict[str, str] = kfp_component.pod_spec
+        if "memory" in pod_spec:
+            container_op.container.set_memory_request(pod_spec["memory"])
+        if "memory_limit" in pod_spec:
+            container_op.container.set_memory_limit(pod_spec["memory_limit"])
+        if "cpu" in pod_spec:
+            container_op.container.set_cpu_request(pod_spec["cpu"])
+        if "cpu_limit" in pod_spec:
+            container_op.container.set_cpu_limit(pod_spec["cpu_limit"])
+        if "gpu" in pod_spec:
             # TODO(yunw)(AIP-2048): Support mixture of GPU from different vendors.
-            gpu_vendor = resource_requirements.get("gpu_vendor", None)
+            gpu_vendor = pod_spec.get("gpu_vendor", None)
             container_op.container.set_gpu_limit(
-                resource_requirements["gpu"],
+                pod_spec["gpu"],
                 vendor=gpu_vendor if gpu_vendor else "nvidia",
             )
+        if "annotations" in pod_spec:
+            for name, value in pod_spec["annotations"].items():
+                container_op.container.add_pod_annotation(name, value)
+        if "labels" in pod_spec:
+            for name, value in pod_spec["labels"].items():
+                container_op.container.add_pod_label(name, value)
```

Comment:
It's details like this in the KFP SDK that are infuriating! Containers don't officially have labels or annotations only pods can 🤦  Not your fault but wish they had scoped these APIs properly.

Reply:
🤦 Thanks for catching this. I put the "Testing not done yet." in description for a reason 😅

Reply:
Ah sorry I figured this was a bug in the KFP SDK but looks like it's actually as I hoped it'd be and you need to use `container_op.add_pod_label` 😅